File: Difficulty: Easy/String Rotated by 2 Places/string-rotated-by-2-places.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    // Function to check if a string can be obtained by rotating
    // another string by exactly 2 places.
    void rotateClockwise(string &s){
        char c = s[s.size() -1];
        int i = s.size() -2;
        while(i >=0){
            s[i+1] = s[i];
            i--;
        }
        s[0] = c;
    
    }
    
    void rotateAnti(string &s){
        char c = s[0];
        int i = 1;
        while(i <= s.size()-1){
            s[i -1] = s[i]; 
            i++;
        }
       s[s.size() -1]= c;
    }
    bool isRotated(string& s1, string& s2) {
        if(s1.size() != s2.size()) return false;
        string clockwise , anticlockwise;
        clockwise = s1;
        rotateClockwise(clockwise);
        rotateClockwise(clockwise);
        if(clockwise ==s2) return true;
        anticlockwise = s1;
        rotateAnti(anticlockwise);
        rotateAnti(anticlockwise);
        if(anticlockwise == s2) return true;
        return false;
    }
};



//{ Driver Code Starts.

int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        string s;
        string b;
        cin >> s >> b;
        cin.ignore();
        Solution obj;
        cout << (obj.isRotated(s, b) == 0 ? "false" : "true") << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Easy/String Rotated by 2 Places/string-rotated-by-2-places.js
//{ Driver Code Starts
// Initial Template for javascript
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split('\n').map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {
        let input_line = readLine().split(' ');
        let str1 = input_line[0];
        input_line = readLine().split(' ');
        let str2 = input_line[0];
        let obj = new Solution();
        if (obj.isRotated(str1, str2))
            console.log("true");
        else
            console.log("false");

        console.log("~");
    }
}
// } Driver Code Ends


// User function Template for javascript

/**
 * @param {string} s1
 * @param {string} s2
 * @returns {boolean}
 */

class Solution {
    // Function to check if a string can be obtained by rotating
    // another string by exactly 2 places.
    isRotated(s1, s2) {
        // code here
        let s3 = s1 + s1;
        let s4 = s2 + s2;
        return s3.indexOf(s2) === 2 || s4.indexOf(s1) === 2
    }
}